### DISCUSSION TITLE
How do I pass url paramaters to an embedded dashboard using the superset-embedded-sdk?

### DISCUSSION THREAD

**Discussion by idrinks** [apache/superset#21956] [Q&A / Help]
I cannot figure out how to pass URL parameters to my embedded dashboard using the superset-embedded SDK.   What I am trying to do is modify the query initial filter parameters of the dataset that the dashboard uses using the Superset SQL Templating URL Parameter function. Please help.

**Comment by DoubleXpresso** [comment]
I am trying to do the same and what have found out that we need to use jinja templating. 

Here are some links : 
https://superset.apache.org/docs/installation/sql-templating/
https://preset.io/blog/intro-jinja-templating-apache-superset/
https://medium.com/geekculture/interactive-dashboards-in-apache-superset-ec4a3bf9bc82

**Reply by alphaadidas** [reply]
But will this work for Embedded Dashboards too?

**Reply by DoubleXpresso** [reply]
To use Jinja templating filters in an embedded Superset dashboard, you will need to ensure that the filter function is defined and registered in the same application context as the embedded dashboard. This means that you will need to define and register the filter function in the application context of your website, rather than in the application context of Superset itself.

One way to do this is to define the filter function in a separate Python module, and import that module into both your website application and your Superset application. You can then register the filter function with the Jinja environment in both applications, so that it is available to both the website and the embedded dashboard.

**Comment by paulobitar** [comment]
> I cannot figure out how to pass URL parameters to my embedded dashboard using the superset-embedded SDK. What I am trying to do is modify the query initial filter parameters of the dataset that the dashboard uses using the Superset SQL Templating URL Parameter function. Please help.

Hi. I'm facing the same problem. Have you solved this ?

**Comment by idrinks** [comment]
Yes, I used the username variable to pass data to my query.   It was the only variable I could find that I could read using the SQL Template.



Warm Regards,


Ivan Drinks Jr.

**Comment by nturri1306** [comment]
hello i've same problem  in js i have
```

    supersetEmbeddedSdk.embedDashboard({
        id: dashboard,
        supersetDomain: domain,
        mountPoint: document.getElementById("test"), // any html element that can contain an iframe
        fetchGuestToken: () => token,
        dashboardUiConfig: { // dashboard UI config: hideTitle, hideTab, hideChartControls, filters.visible, filters.expanded (optional)
            hideTitle: true,
            filters: {
                  expanded: true
                 },
            urlParams: {
             patient : 592738
        },
       },
    });
```

and sql
`
SELECT * from  "http://172.16.14.191:7080/api/fhir/Patient?_id={{url_param('patient')}}"`

but the parameter is not arrive

**Reply by KrishnaLsn** [reply]
Hi @nturri1306, Did you find any solution ???
I am also trying to find a way to use URL params filtering capability through SDK.

Also, I am using TS and there is no param like urlParams in EmbedDashboardParams. What I meant to say is, we can't pass url params like a param to embed sdk. Hope, it helps

 urlParams: {
             patient : 592738
        },

Thank you

**Reply by okayakdeniz** [reply]
Unfortunatelly, still open issue now. İ have face that two people suggest this feature pull request. Many people wait in the slack. We may push superset team more

**Reply by FahmiChaar** [reply]
Hi,
You can use this fork of the sdk to pass url_params
- change this
`<script src="https://unpkg.com/@superset-ui/embedded-sdk"></script>`
- to this
`<script src="https://unpkg.com/superset-embedded-sdk"></script>`

- then add urlParams object

```
dashboardUiConfig: {
  ...,
  urlParams: {
      patient : 592738
  }
}
```

**Reply by gaeljw** [reply]
@FahmiChaar is there any open PR to add your work to the official embedded SDK?

I found the following ones:
- https://github.com/apache/superset/pull/24408
- https://github.com/apache/superset/pull/20259
- https://github.com/apache/superset/pull/23344

**Reply by ghost** [reply]
@FahmiChaar 
Hello,
before I use this one <script src="https://unpkg.com/@superset-ui/embedded-sdk"></script>
its working to open the UI page on my web but not working with url params
but when I use this one <script src="https://unpkg.com/superset-embedded-sdk"></script>
its working with url params but does not open the UI
